src/environmentd/tests/server.rs

```diff
@@ -359,14 +359,17 @@ fn test_cancel_long_running_query() {
     let mut client = server.connect(postgres::NoTls).unwrap();
     let cancel_token = client.cancel_token();
 
+    client.batch_execute("CREATE TABLE t (i INT)").unwrap();
+
     thread::spawn(move || {
-        // Abort the query after 2s.
-        thread::sleep(Duration::from_secs(2));
-        let _ = cancel_token.cancel_query(postgres::NoTls);
+        // Repeatedly attempt to abort the query because we're not sure exactly
+        // when the SELECT will arrive.
+        loop {
+            thread::sleep(Duration::from_secs(1));
+            let _ = cancel_token.cancel_query(postgres::NoTls);
+        }
```

Comment:
I think this thread will leak after the test finishes, which seems not great. Also slight chance that it cancels the query on 379 incorrectly. I think we need a little synchronization to make sure the thread shuts down on L378. Probably easiest to switch it to a Tokio task, abort it on L378, and then await the handle, to make sure it's good and truly gone?

Reply:
Ah yes. Used not tokio instead because it seemed easier than dealing with a runtime.